pkg/datarepair/checker/checker.go

```diff
@@ -101,14 +105,37 @@ func (c *checker) identifyInjuredSegments(ctx context.Context) (err error) {
 					return Error.New("error getting offline nodes %s", err)
 				}
 				numHealthy := len(nodeIDs) - len(missingPieces)
-				if int32(numHealthy) < pointer.Remote.Redundancy.RepairThreshold {
+				if (int32(numHealthy) >= pointer.Remote.Redundancy.MinReq) && (int32(numHealthy) < pointer.Remote.Redundancy.RepairThreshold) {
 					err = c.repairQueue.Enqueue(&pb.InjuredSegment{
 						Path:       string(item.Key),
 						LostPieces: missingPieces,
 					})
 					if err != nil {
 						return Error.New("error adding injured segment to queue %s", err)
 					}
+				} else if int32(numHealthy) < pointer.Remote.Redundancy.MinReq {
+					// make an entry in to the irreparabledb table
+					rmtSegInfo := &pb.RmtSegInfo{
+						Key:                item.Key,
+						Val:                item.Value,
+						LostPiecesCount:    int64(len(missingPieces)),
+						RepairUnixSec:      time.Now().Unix(),
+						RepairAttemptCount: int64(1),
+					}
+
+					apiKey, ok := auth.GetAPIKey(ctx)
```

Comment:
This APIKey check is not necessary. Line ` c.irrdb.Put(ctx, putReq)` will throw an error if APIKey will be invalid or missing.

Reply:
Done